conjure-java-jersey-server/src/main/java/com/palantir/conjure/java/server/jersey/RemoteExceptionMapper.java

```diff
@@ -54,64 +54,35 @@
 
     @Override
     public Response toResponseInner(RemoteException exception) {
-        ErrorType errorType = mapErrorType(exception);
-        Response.ResponseBuilder builder = Response.status(errorType.httpErrorCode());
-
-        try {
-            SerializableError error = SerializableError.builder()
-                    .errorName(errorType.name())
-                    .errorCode(errorType.code().toString())
-                    .errorInstanceId(exception.getError().errorInstanceId())
-                    .build();
-            builder.type(MediaType.APPLICATION_JSON);
-            builder.entity(error);
-        } catch (RuntimeException e) {
-            log.warn(
-                    "Unable to translate exception to json",
-                    SafeArg.of("errorInstanceId", exception.getError().errorInstanceId()),
-                    SafeArg.of("errorName", exception.getError().errorName()),
-                    e);
-            // simply write out the exception message
-            builder.type(MediaType.TEXT_PLAIN);
-            builder.entity("Unable to translate exception to json. Refer to the server logs with this errorInstanceId: "
-                    + exception.getError().errorInstanceId());
-        }
-        return builder.build();
-    }
-
-    private ErrorType mapErrorType(RemoteException exception) {
-        Status status = Status.fromStatusCode(exception.getStatus());
-
-        if (status.getStatusCode() == 401) {
-            log.info(
-                    "Encountered a remote unauthorized exception."
-                            + " Mapping to a default unauthorized exception before propagating",
-                    SafeArg.of("errorInstanceId", exception.getError().errorInstanceId()),
-                    SafeArg.of("errorName", exception.getError().errorName()),
-                    SafeArg.of("statusCode", status.getStatusCode()),
-                    exception);
-
-            return ErrorType.UNAUTHORIZED;
-        } else if (status.getStatusCode() == 403) {
+        if (exception.getStatus() == 401 || exception.getStatus() == 403) {
```

Comment:
We could limit this a bit further here and only propagate remote exceptions whose error code matches the corresponding generic `ErrorType`. But I don't think this would make a significant difference in practice. If we receive a 401 or 403 error it's almost certainly going to be with those error codes.

Reply:
Most likely scenario is a new `Code` (several existing codes share status codes). In that case we wouldn't want older services to fail to propagate that data :+1: